cupy/core/_routines_math.pyx

```diff
@@ -393,6 +396,17 @@ cdef ndarray scan(ndarray a, op, dtype=None, ndarray out=None):
     if a._shape.size() != 1:
         raise TypeError('Input array should be 1D array.')
 
+    for backend in _backend._routine_backends:
+        if backend == _backend.BACKEND_CUB:
+            # result will be None if the scan is not compatible with CUB
+            if op == scan_op.SCAN_SUM:
+                cub_op = cub.CUPY_CUB_CUMSUM
+            else:
+                cub_op = cub.CUPY_CUB_CUMPROD
+            res = cub.cub_scan(out, cub_op)
+            if res is not None:
+                return out
```

Comment:
We probably need this here (for fast indexing) as well as in `scan_core()` (where this came from), because `cupy/math/sumprod.py` calls the latter directly. Moreover, this early appearance does not take care:
1. copy input: scan is an in-place operation (whether using CUB or not), so the input `a` must be copied to `out`.
2. ravel input: CUB scan needs a 1D contiguous input.
3. axis: CUB only handles non-batched scans (`axis is None`).

So at least for `cupy.cumsum()` and `cupy.cumprod()` this won't work.

Reply:
Fix. Thanks!

Reply:
Thank you.